graylog2-server/src/main/java/org/graylog2/Configuration.java

```diff
@@ -158,6 +159,11 @@
     @Parameter(value = "deactivated_builtin_authentication_providers", converter = StringSetConverter.class)
     private Set<String> deactivatedBuiltinAuthenticationProviders = Collections.emptySet();
 
+    // Defaults to TLS protocols that are currently considered secure
+    @Parameter(value = "enabled_tls_protocols", converter = StringSetConverter.class)
+    private Set<String> enabledTlsProtocols = ImmutableSet.of("TLSv1.2", "TLSv1.3");
```

Comment:
I noticed that the default we use in 3.1 and earlier is `TLSv1.0`, `TLSv1.1` and `TLSv1.2`. Changing this to `TLSv1.2` and `TLSv1.3` only would be a breaking change and might result in broken user and customer setups. So I think we need to enable all previous defaults to avoid any issues.

We should probably document that people should switch to more secure ones. With the next major release (v4) we can change the default.

Reply:
I disagree. TLS 1.0 and 1.1 are known to have security issues. We should be shipping with a secure default.
TLS 1.2 is supported in major browsers since ~2013 and is the de-facto standard. 94% global usage in 2018. https://www.ixiacom.com/company/blog/current-state-tls  

I think the fallout will be very small. We can make the change (and workaround) prominent in the 3.2 release notes.
At the very least, we should disable TLS 1.0 (since that one is not even PCI DSS compliant anymore)
https://blog.pcisecuritystandards.org/are-you-ready-for-30-june-2018-sayin-goodbye-to-ssl-early-tls

Reply:
cf https://portswigger.net/daily-swig/the-end-is-nigh-browser-makers-ditch-support-for-aging-tls-1-0-1-1-protocols

Reply:
Okay, we talked about this offline. I assumed this changes should be backported to 3.1, but that isn't the case because there is a workaround. (will be documented in #6444)